theano/tensor/var.py

```diff
@@ -499,25 +503,22 @@ def includes_bool(args_el):
                                     'To use a boolean mask, convert the mask to '
                                     'a NumPy array first, e.g., '
                                     'tensor[numpy.array([True, False])].')
-                tmp_args.append(arg)
-        args = tuple(tmp_args)
+                index_dim_count += 1
+
+        # Check if the number of dimensions isn't too large.
+        if self.ndim < index_dim_count:
+            raise IndexError('too many indices for array')
 
         # Convert an Ellipsis if provided into an appropriate number of
         # slice(None).
-        ellipses = [i
-                    for i, index in enumerate(args)
-                    if index is Ellipsis]
         if len(ellipses) > 1:
             raise IndexError(
                 "an index can only have a single Ellipsis (`...`)")
         elif len(ellipses) == 1:
-            new_axes = sum(1
-                           for index in args
-                           if index is np.newaxis)  # numpy.newaxis is None
             ellipsis_at = ellipses[0]
             args = list(args)
             args[ellipsis_at: ellipsis_at + 1] = (
-                [slice(None)] * (self.ndim - (len(args) - 1 - new_axes)))
+                [slice(None)] * (self.ndim - index_dim_count))
```

Comment:
Yes, that's on purpose, because the boolean arrays are treated differently from integer arrays (see my other comment). An integer array counts for 1 dimension no matter what shape it is, but a boolean array counts for ndim dimensions. The loop a few lines before this does that counting.